screens/SendSetupScreen.js

```diff
@@ -331,7 +346,99 @@ const SendSetupScreen = ({
     }
   };
 
+  const handleDeepLink = deepLinkParams => {
+    const { address, tokenId, symbol } = deepLinkParams;
+    const type = getType(address);
+    if (typeof type === "object") {
+      setErrors(["Invalid Address"]);
+    }
+    if (type === "cashaddr") {
+      return prefillBCH(deepLinkParams);
+    }
+    if (type === "slpaddr") {
+      if (tokenId !== undefined && symbol === undefined) {
+        setErrors(["Missing Token Information"]);
+        redirectHome(navigation);
+      } else {
+        prefillSLP(deepLinkParams);
+      }
+    }
+  };
+
+  const prefillSLP = deepLinkParams => {
+    const {
+      address,
+      amount1,
+      amount,
+      tokenAmount,
+      label,
+      symbol,
+      tokenId
+    } = deepLinkParams;
+
+    const hasTokenAmount = address !== undefined && tokenAmount !== undefined;
+    const noTokenBalance = tokenId !== undefined && symbol === undefined;
+    if (noTokenBalance) {
+      setErrors(["Insufficient Token Balance"]);
+      redirectHome(navigation);
+    }
+    if (tokenId === undefined) {
+      setErrors(["Missing Token Type"]);
+      redirectHome(navigation);
+    }
+
+    if (hasTokenAmount) {
+      try {
+        setSendAmountCrypto(tokenAmount);
+        // const numericalAmount = parseAmount(tokenAmount);
+        setAmountType("crypto");
+        setSendAmount(tokenAmount);
+        goNextStep();
+      } catch (error) {
+        setErrors(["Invalid Amount"]);
+        redirectHome(navigation);
+      }
+    }
+    setToAddress(address);
+  };
+
+  const prefillBCH = deepLinkParams => {
+    const { address, amount, label, message } = deepLinkParams;
+
+    const hasAmount = address !== undefined && amount !== undefined;
+    navigation.state.params.symbol = "BCH";
+    if (hasAmount) {
+      try {
+        setSendAmountCrypto(amount);
+        const numericalAmount = parseAmount(amount);
+        setAmountType("crypto");
+        setSendAmount(amount);
+        setSendAmountFiat(
+          fiatRate
+            ? (fiatRate * (numericalAmount || 0)).toFixed(
+                currencyDecimalMap[fiatCurrency]
+              )
+            : 0
+        );
+
+        goNextStep();
+      } catch (error) {
+        setErrors(["invalid amount"]);
+      }
+    }
+
+    setToAddress(address);
+  };
+
   useEffect(() => {
+    const deepLinkParams =
+      navigation.state.params !== undefined ? navigation.state.params : "";
+    const hasDeepLinkParams = deepLinkParams !== "";
+
+    if (hasDeepLinkParams !== undefined && fiatRate !== undefined) {
```

Comment:
fiatRate will likely often be undefined at this point, do we want it to stop the deeplink / is that data really needed at this point?